### PR TITLE
rebase 1.17.0 rc.2

### PR DESCRIPTION
**Pull request by p0lyn0mial**
TODO in this PR
- [x] sync with https://github.com/openshift/origin/commits/master/vendor/k8s.io (add missing commits)
- [x] open a bugzilla for the opencontainers/runc problem for the node team - https://bugzilla.redhat.com/show_bug.cgi?id=1784793
- [x] you may not knowingly disable conformance tests.
- [ ] announce the code freeze
- [ ] add the special commit at the end (rebase point) (assign @mfojtik)

TODO after
- [ ] pick https://github.com/kubernetes/kubernetes/pull/86256 or at least https://github.com/kubernetes/kubernetes/pull/86252 (@openshift/openshift-team-developer-productivity-embedded)
- [x] fix https://bugzilla.redhat.com/show_bug.cgi?id=1784074
- [ ] pick https://github.com/kubernetes/kubernetes/pull/86212 (@mfojtik) (@openshift/openshift-team-developer-productivity-embedded)
- [ ] make sure we don't serve extensions/v1beta1 and beta levels of apps by default
- [ ] remove `horizontalpodautoscalers` and `securitycontextconstraints` from the gc patch https://github.com/openshift/kubernetes/blob/origin-4.4-kubernetes-1.16.2/cmd/kube-controller-manager/app/patch_gc.go#L40

This PR disables the following tests which were disabled before as part of `sig-cluster-lifecycle` suite, see https://github.com/kubernetes/kubernetes/commit/0b3d50b6dccdc4bbd0b3e411c648b092477d79ac#diff-3b1910d08fb8fd8b32956b5e264f87cb

```
"[k8s.io] [sig-cloud-provider-gcp] Downgrade [Feature:Downgrade] cluster downgrade should maintain a functioning cluster [Feature:ClusterDowngrade] [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                                             
"[k8s.io] [sig-cloud-provider-gcp] Ports Security Check [Feature:KubeletSecurity] should not be able to proxy to cadvisor port 4194 using proxy subresource [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                                     
"[k8s.io] [sig-cloud-provider-gcp] Ports Security Check [Feature:KubeletSecurity] should not be able to proxy to the readonly kubelet port 10255 using proxy subresource [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                        
"[k8s.io] [sig-cloud-provider-gcp] Ports Security Check [Feature:KubeletSecurity] should not have port 10255 open on its all public IP addresses [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                                                
"[k8s.io] [sig-cloud-provider-gcp] Ports Security Check [Feature:KubeletSecurity] should not have port 4194 open on its all public IP addresses [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                                                 
"[k8s.io] [sig-cloud-provider-gcp] Upgrade [Feature:Upgrade] cluster upgrade should maintain a functioning cluster [Feature:ClusterUpgrade] [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                                                     
"[k8s.io] [sig-cloud-provider-gcp] Upgrade [Feature:Upgrade] master upgrade should maintain a functioning cluster [Feature:MasterUpgrade] [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                                                       
"[k8s.io] [sig-cloud-provider-gcp] Upgrade [Feature:Upgrade] node upgrade should maintain a functioning cluster [Feature:NodeUpgrade] [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                                                           
"[k8s.io] [sig-cloud-provider-gcp] etcd Upgrade [Feature:EtcdUpgrade] etcd upgrade should maintain a functioning cluster [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                                                                        
"[k8s.io] [sig-cloud-provider-gcp] gpu Upgrade [Feature:GPUUpgrade] cluster downgrade should be able to run gpu pod after downgrade [Feature:GPUClusterDowngrade] [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                               
"[k8s.io] [sig-cloud-provider-gcp] gpu Upgrade [Feature:GPUUpgrade] cluster upgrade should be able to run gpu pod after upgrade [Feature:GPUClusterUpgrade] [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                                     
"[k8s.io] [sig-cloud-provider-gcp] gpu Upgrade [Feature:GPUUpgrade] master upgrade should NOT disrupt gpu pod [Feature:GPUMasterUpgrade] [Suite:openshift/conformance/parallel] [Suite:k8s]"                                                                                                                        
"[k8s.io] [sig-cloud-provider-gcp] kube-proxy migration [Feature:KubeProxyDaemonSetMigration] Downgrade kube-proxy from a DaemonSet to static pods should maintain a functioning cluster [Feature:KubeProxyDaemonSetDowngrade] [Suite:openshift/conformance/parallel] [Suite:k8s]"                                  
"[k8s.io] [sig-cloud-provider-gcp] kube-proxy migration [Feature:KubeProxyDaemonSetMigration] Upgrade kube-proxy from static pods to a DaemonSet should maintain a functioning cluster [Feature:KubeProxyDaemonSetUpgrade] [Suite:openshift/conformance/parallel] [Suite:k8s]"
```

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
requires https://github.com/openshift/origin/pull/24305

**Comment by p0lyn0mial** [comment]
requires https://github.com/openshift/origin/pull/24304

File: glide.yaml [hunk 1]
```diff
@@ -8,99 +8,109 @@ import:
...
 
 # pin for etcd
 - package: google.golang.org/grpc
-  version: v1.23.0
+  version: v1.23.1
 
-# etcd v3.3.17 carry until 4.4
 - package: github.com/coreos/etcd
-  version: v3.3.17
+  version: v3.3.10
```
Comment (deads2k) [inline glide.yaml]: this backleveling isn't correct.  Get up with @hexfusion to find the right level, but at the very least don't backlevel it.

File: test/e2e/upgrade/upgrade.go [hunk 2]
```diff
@@ -115,8 +115,8 @@ var _ = g.Describe("[Disruptive]", func() {
 			client := configv1client.NewForConfigOrDie(config)
 			dynamicClient := dynamic.NewForConfigOrDie(config)
 
-			upgCtx, err := getUpgradeContext(client, lifecycle.GetUpgradeTarget(), lifecycle.GetUpgradeImage())
-			framework.ExpectNoError(err, "determining what to upgrade to version=%s image=%s", lifecycle.GetUpgradeTarget(), lifecycle.GetUpgradeImage())
+			upgCtx, err := getUpgradeContext(client, gcp.GetUpgradeTarget(), gcp.GetUpgradeImage())
```
Comment (deads2k) [inline test/e2e/upgrade/upgrade.go]: I'm surprised this became cloud specific.  You sure?

File: test/extended/etcd/etcd_storage_path.go [hunk 3]
```diff
@@ -275,7 +275,19 @@ func testEtcd3StoragePath(t g.GinkgoTInterface, kubeConfig *restclient.Config, e
 		gvr("scheduling.k8s.io", "v1alpha1", "priorityclasses"),
 		gvr("settings.k8s.io", "v1alpha1", "podpresets"),
 		gvr("storage.k8s.io", "v1alpha1", "volumeattachments"),
-		gvr("discovery.k8s.io", "v1alpha1", "endpointslices"),
+		gvr("extensions", "v1beta1", "deployments"),
```
Comment (deads2k) [inline test/extended/etcd/etcd_storage_path.go]: these look wrong.  Are we enabling endpoints that need to be shut off?  Can we function with them shut off

File: test/extended/util/test.go [hunk 4]
```diff
@@ -384,6 +385,7 @@ var (
 			`\[Feature:Audit\]`,                         // Needs special configuration
 			`\[Feature:LocalStorageCapacityIsolation\]`, // relies on a separate daemonset?
 			`\[sig-cluster-lifecycle\]`,                 // cluster lifecycle test require a different kind of upgrade hook.
+			`\[sig-cloud-provider-gcp\]`,                // these test require a different configuration - note that GCE tests from the sig-cluster-lifecycle were moved to the sig-cloud-provider-gcpcluster lifecycle see https://github.com/kubernetes/kubernetes/commit/0b3d50b6dccdc4bbd0b3e411c648b092477d79ac#diff-3b1910d08fb8fd8b32956b5e264f87cb
```
Comment (deads2k) [inline test/extended/util/test.go]: please make a list of which tests this disables in the description of this PR.

**Comment by deads2k** [comment]
@p0lyn0mial you can't knowingly disable conformance tests.  We can ship a non-conformant distro, so the GC issue needs to be chased down.

File: glide.yaml (hunk 1 above)
Comment (p0lyn0mial) [reply glide.yaml]: this is exactly what upstream uses https://github.com/kubernetes/kubernetes/blob/v1.17.0-rc.2/go.mod#L231

also `github.com/coreos/etcd` was replaced by `go.etcd.io/etcd`

Comment (p0lyn0mial) [reply glide.yaml]: I remember some issues in that area I can try to delete it.

File: test/extended/etcd/etcd_storage_path.go (hunk 3 above)
Comment (p0lyn0mial) [reply test/extended/etcd/etcd_storage_path.go]: I think that https://github.com/openshift/origin/blob/015a65f45fc0889bb1ac23af430241844a50476a/test/extended/etcd/etcd_storage_path.go#L267

at runtime extensions/v1beta1 are disabled.

File: test/e2e/upgrade/upgrade.go (hunk 2 above)
Comment (p0lyn0mial) [reply test/e2e/upgrade/upgrade.go]: yes, https://github.com/kubernetes/kubernetes/commit/0b3d50b6dccdc4bbd0b3e411c648b092477d79ac#diff-3699a4098a3cce570a8d7d0a3a4b5574

File: test/extended/util/test.go (hunk 4 above)
Comment (p0lyn0mial) [reply test/extended/util/test.go]: done.

File: test/extended/etcd/etcd_storage_path.go (hunk 3 above)
Comment (p0lyn0mial) [reply test/extended/etcd/etcd_storage_path.go]: without it, the test fails
```
fail [github.com/openshift/origin/test/extended/etcd/etcd_test_runner.go:72]: test failed:
etcd data does not match the types we saw:
in etcd data but not seen:
[
	apps/v1beta1, Resource=controllerrevisions 
	extensions/v1beta1, Resource=podsecuritypolicies 
	extensions/v1beta1, Resource=replicasets 
	extensions/v1beta1, Resource=deployments 
	apps/v1beta1, Resource=deployments 
	apps/v1beta2, Resource=replicasets 
	apps/v1beta2, Resource=statefulsets 
	apps/v1beta2, Resource=deployments 
	extensions/v1beta1, Resource=networkpolicies 
	apps/v1beta2, Resource=daemonsets 
	apps/v1beta2, Resource=controllerrevisions 
	apps/v1beta1, Resource=statefulsets 
	extensions/v1beta1, Resource=daemonsets]
seen but not in etcd data:
[]
```

for example https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24286/pull-ci-openshift-origin-master-e2e-aws-serial/11496

Comment (p0lyn0mial) [reply test/extended/etcd/etcd_storage_path.go]: `etcd data` is coming from https://github.com/openshift/kubernetes/blob/origin-4.4-kubernetes-1.17.0-rc.2/test/integration/etcd/data.go#L39

File: glide.yaml (hunk 1 above)
Comment (p0lyn0mial) [reply glide.yaml]: @hexfusion is away until `1/2` we need to make a decision.

**Comment by p0lyn0mial** [comment]
@adambkaplan  FYI I think that I am going to disable the following tests as they are extremely unstable

```
[Feature:Builds][Slow] update failure status Build status OutOfMemoryKilled should contain OutOfMemoryKilled failure reason and message [Suite:openshift] 

[Feature:OpenShiftControllerManager] TestDockercfgTokenDeletedController [Suite:openshift/conformance/parallel] 

[Feature:Platform] Service Catalog should check basic usages: OCP-24062, OCP-24049, OCP-15600 [Suite:openshift/conformance/parallel]
```

**Comment by bparees** [comment]
service catalog is not @adambkaplan.  you'll want @shawn-hurley for that.

and the oomkill test is failing due to an rhcos bug that's being resolved.

File: glide.yaml (hunk 1 above)
Comment (p0lyn0mial) [reply glide.yaml]: we decided to revert it back to `v3.3.17`

**Comment by deads2k** [comment]
/lgtm
/hold

holding for at least one day.

File: glide.yaml [hunk 5]
```diff
@@ -8,73 +8,73 @@ import:
...
-  version: origin-4.4-kubernetes-1.16.2
+  version: origin-4.4-kubernetes-1.17.0-rc.2
 - package: k8s.io/sample-controller
   repo:    https://github.com/openshift/kubernetes-sample-controller.git
-  version: origin-4.4-kubernetes-1.16.2
+  version: origin-4.4-kubernetes-1.17.0-rc.2
 - package: k8s.io/kubernetes
-  repo:    https://github.com/openshift/kubernetes.git
-  version: origin-4.4-kubernetes-1.16.2
+  repo:    https://github.com/p0lyn0mial/kubernetes.git
```
Comment (soltysh) [inline glide.yaml]: That shouldn't be here, https://github.com/openshift/kubernetes/pull/102 already merged.

File: test/extended/include.go [hunk 6]
```diff
@@ -12,18 +12,12 @@ import (
 	_ "k8s.io/kubernetes/test/e2e/instrumentation"
 	_ "k8s.io/kubernetes/test/e2e/kubectl"
 
-	// _ "k8s.io/kubernetes/test/e2e/lifecycle"
```
Comment (soltysh) [inline test/extended/include.go]: Sync with @smarterclayton before you remove it from here, I'm not sure if these need to brought back or not, I see that he added them commented out from the start, so better to double check.

Comment (p0lyn0mial) [reply test/extended/include.go]: I think that these tests were disabled 2 years ago :) https://github.com/openshift/origin/commit/76d0cb1bbaf44761638f4886614ee8a01c2ae749

File: glide.yaml (hunk 5 above)
Comment (p0lyn0mial) [reply glide.yaml]: fixed!, thx.

**Comment by smarterclayton** [comment]
/hold

Until I see some soak coverage

**Comment by smarterclayton** [comment]
Panic during an upgrade test:

```
2019/12/19 03:27:04 Container setup in pod e2e-azure-upgrade completed successfully
53
panic: runtime error: invalid memory address or nil pointer dereference
54
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x127b1c6]
55
56
goroutine 1 [running]:
57
github.com/openshift/origin/vendor/github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).CurrentSpecSummary(...)
58
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/github.com/onsi/ginkgo/internal/specrunner/spec_runner.go:211
59
github.com/openshift/origin/vendor/github.com/onsi/ginkgo/internal/suite.(*Suite).CurrentRunningSpecSummary(0xc00029b400, 0x121a, 0x121b)
60
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/github.com/onsi/ginkgo/internal/suite/suite.go:105 +0x26
61
github.com/openshift/origin/vendor/github.com/onsi/ginkgo.CurrentGinkgoTestDescription(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
62
```

https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/logs/release-openshift-origin-installer-e2e-azure-upgrade/2

Did you drop a patch accidentally?

**Comment by smarterclayton** [comment]
Disruptive tests broke, please sync with Eric and get the bastion updated:

https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24286/pull-ci-openshift-origin-master-e2e-aws-disruptive/35

**Comment by smarterclayton** [comment]
test/conformance-k8s.sh needs to be bumped to point to 1.17, then invoke the job with test e2e-conformance-k8s

**Comment by p0lyn0mial** [comment]
> Disruptive tests broke, please sync with Eric and get the bastion updated:

opened https://github.com/eparis/ssh-bastion/pull/15 @eparis PTAL

**Comment by p0lyn0mial** [comment]
> /test e2e-conformance-k8s

ci/prow/e2e-conformance-k8s — Job succeeded.

**Comment by p0lyn0mial** [comment]
> Did you drop a patch accidentally?

seems like I forgot to add one, I don't know how the other tests can run without it. @smarterclayton could you run `e2e-azure-upgrade` one more time?

**Comment by smarterclayton** [comment]
https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/logs/release-openshift-origin-installer-e2e-azure-upgrade/3
https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade-4.4/12

**Comment by smarterclayton** [comment]
Here are the skew tests (1.17 control plane with 1.16 kubelets)

https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24286/pull-ci-openshift-origin-master-e2e-gcp/4833
https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24286/pull-ci-openshift-origin-master-e2e-aws/13340

**Comment by smarterclayton** [comment]
The OOMFailure on build is https://bugzilla.redhat.com/show_bug.cgi?id=1785326

**Comment by adambkaplan** [comment]
BZ opened for builds OOMKilled tests failing: https://bugzilla.redhat.com/show_bug.cgi?id=1785326

There is an RHCOS fix which was _supposed_ to have landed already - if this either a) hasn't landed, or b) did land, but didn't fix this issue, I am OK with skipping the test.

**Comment by adambkaplan** [comment]
#24332 disables the build OOM test

**Comment by deads2k** [comment]
`TestUniformDistribution` can be straight skipped

**Comment by deads2k** [comment]
new skew:

https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24286/pull-ci-openshift-origin-master-e2e-aws/13342
https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24286/pull-ci-openshift-origin-master-e2e-aws/13341

**Comment by smarterclayton** [comment]
conmon fix is in 44.81.201912191631.0, which is in 4.4 as of 3 hours ago.

**Comment by deads2k** [comment]
/hold cancel
/lgtm
/override ci/prow/e2e-gcp-builds
for known oom problems

/override ci/prow/e2e-aws-disruptive
for known bastion issues

see if we instamerge

File: glide.yaml (hunk 1 above)
Comment (hexfusion) [reply glide.yaml]: thanks 100% agree

**Pull request merged**